npm/private/npm_translate_lock_helpers.bzl

```diff
@@ -498,23 +497,12 @@ Either remove this patch file if it is no longer needed or change its key to mat
 
 ################################################################################
 def _collect_deps(packages, package_info):
-    # Quick-exit for packages with no optional dependencies
-    if not package_info["optional_dependencies"]:
-        return package_info["dependencies"], None
-
     constraints = {}
-    odeps = {}
-
-    for dep_name, dep_key in package_info["optional_dependencies"].items():
-        if dep_name in package_info["dependencies"]:
```

Comment:
Previously if a dep was in `optional_dependencies` + `dependencies` we would not add the constraints. I think instead it should not be listed in `optional_dependencies` in the first place, which seems to already be the case